waiter/src/waiter/handler.clj

```diff
@@ -1062,9 +1060,7 @@
       (when-not (seq service-description-template)
         (throw (ex-info "Unable to load description for token" {:log-level :info :status http-404-not-found :token token})))
       (let [auth-user (:authorization/user request)
-            service-id (-> service-description-template
-                           (sd/assoc-run-as-requester-fields auth-user)
-                           service-description->service-id)
+            service-id (request->consent-service-id request)
```

Comment:
Why is service-id computed separately so many places???

Reply:
The two places it is computed in the consent worflow:

1. Computed to include the HTML form we generate to render the consent form
2. Computed to confirm that the service-ID sent in the form submission matches the service we are trying to launch.